worlds/tits_the_3rd/client/memory_io.py

```diff
@@ -273,3 +274,102 @@ def read_flag(self, flag_number: int):
         data = self._read_bytes(flag_byte_offset, 1)
         flag_value = (data >> flag_bit) & 1
         return bool(flag_value)
+
+    def get_scena_offset(self, scena_id: bytes):
+        """Get and verify a cached scena offset. Find and cache if it does not exist"""
+        if scena_id not in self.scena_offsets:
+            address = self.tits_the_3rd_mem.pattern_scan_all(scena_id)
+            if not address:
+                logger.error("Could not find give item function. Is the game running?")
+                return None
+            self.scena_offsets[scena_id] = address
+
+        if self.tits_the_3rd_mem.read_bytes(self.scena_offsets[scena_id], 2) != b"AP":
```

Comment:
Suggest starting this if case with `del self.scena_offsets[scena_id]`. That way if it doesn't find the function, we don't still have a corrupt mapping in the cache.